Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/UI/RecordingServiceVisual.cs

```diff
@@ -93,17 +93,18 @@ protected void Awake()
                 Debug.LogError("Error: Recording service not set for RecordingServiceVisual");
                 return;
             }
-
-            _recordButton.onClick.AddListener(OnRecordClick);
-            _previewButton.onClick.AddListener(OnPreviewClick);
         }
 
         protected void Update()
         {
             if (_recordingService != null &&
-                _previewButton != null)
+                _recordButton != null &&
+                _recordButton.gameObject != null &&
+                _previewButton != null &&
+                _previewButton.gameObject != null)
             {
-                _previewButton.gameObject?.SetActive(_recordingService.IsRecordingAvailable());
+                bool showPreviewButton = _recordingService.IsRecordingAvailable() && _recordButton.gameObject.activeSelf;
```

Comment:
>_recordButton.gameObject.activeSelf [](start = 85, length = 35)

Does this signify "not currently recording", is there a better field that could be used?

Reply:
No, technically this checks "Has any recording been made". I think that parenting these buttons to toggle their visibility will improve this experience. But I should probably move away from continually calling this api. Once it's true, itll always be true.

Reply:
scratch that, once its true for android it will always be true. For ios it may vary. I'll update the android recording service to reflect this, but will continual check this value each update